includes/forms/class-wp-job-manager-form-submit-job.php

```diff
@@ -173,7 +173,7 @@ public function init_fields() {
 					'label'       => __( 'Job type', 'wp-job-manager' ),
 					'type'        => $job_type,
 					'required'    => true,
-					'placeholder' => '',
+					'placeholder' => __( 'Choose job type&hellip;', 'wp-job-manager' ),
```

Comment:
Plus, you need to translate hellip. Your method might not work for non-latin language.
It's how WP do it too.
https://github.com/WordPress/WordPress/blob/7ad554f955a1eac48c0361c9997af5cbe8c49dd8/wp-content/themes/twentyseventeen/searchform.php#L19